Services.Test/Diagnostics/DiagnosticsLoggerTest.cs

```diff
@@ -0,0 +1,105 @@
+    {
+        private const string DIAGNOSTICS_SERVICE_URL = @"http://diagnostics";
+
+        private readonly Mock<IHttpClient> mockHttpClient;
+
+        public DiagnosticsLoggerTest()
+        {
+            this.mockHttpClient = new Mock<IHttpClient>();
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ShouldLogServiceStart()
+        {
+            // Arrange
+            var response = new HttpResponse();
+
+            DiagnosticsLogger diagnosticsLogger = new DiagnosticsLogger(
+                this.mockHttpClient.Object,
+                new ServicesConfig
+                {
+                    DiagnosticsEndpointUrl = DIAGNOSTICS_SERVICE_URL
+                });
+
+            this.mockHttpClient
+                .Setup(x => x.PostAsync(It.IsAny<IHttpRequest>()))
+                .ReturnsAsync(response);
+
+            // Act
+            IHttpResponse result = diagnosticsLogger.LogServiceStartAsync("test").Result;
+
+            // Assert - Testing to see if the logic in the function is working fine. 
+            // So, asserting if the expected response and actual responses are similar.
+            Assert.Equal(response, result);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ShouldLogServiceHeartbeat()
+        {
+            // Arrange
+            var response = new HttpResponse();
+
+            DiagnosticsLogger diagnosticsLogger = new DiagnosticsLogger(
+                this.mockHttpClient.Object,
+                new ServicesConfig
+                {
+                    DiagnosticsEndpointUrl = DIAGNOSTICS_SERVICE_URL
+                });
+
+            this.mockHttpClient
+                .Setup(x => x.PostAsync(It.IsAny<IHttpRequest>()))
+                .ReturnsAsync(response);
+
+            // Act
+            IHttpResponse result = diagnosticsLogger.LogServiceHeartbeatAsync().Result;
+
+            // Assert - Testing to see if the logic in the function is working fine. 
+            // So, asserting if the expected response and actual responses are similar.
+            Assert.Equal(response, result);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ShouldLogServiceError()
+        {
+            // Arrange
+            var response = new HttpResponse();
+
+            DiagnosticsLogger diagnosticsLogger = new DiagnosticsLogger(
+                this.mockHttpClient.Object,
+                new ServicesConfig
+                {
+                    DiagnosticsEndpointUrl = DIAGNOSTICS_SERVICE_URL
+                });
+
+            this.mockHttpClient
+                .Setup(x => x.PostAsync(It.IsAny<IHttpRequest>()))
+                .ReturnsAsync(response);
+
+            // Act
+            // Logging service error sending just a message string
+            IHttpResponse logging_service_error_message = diagnosticsLogger.LogServiceErrorAsync("testmessage").Result;
```

Comment:
code style: we don't use snake_case, please use camelCase as for the other variables